Cubical/Categories/Constructions/Coproduct.agda

```diff
@@ -0,0 +1,128 @@
+-- The coproduct of two categories, with its universal property
+{-# OPTIONS --safe #-}
+module Cubical.Categories.Constructions.Coproduct where
+
+open import Cubical.Foundations.Prelude
+open import Cubical.Foundations.Path
+open import Cubical.Foundations.HLevels
+open import Cubical.Foundations.Function
+
+open import Cubical.Data.Sum as Sum hiding (rec; elim; _⊎_)
+open import Cubical.Data.Empty as Empty hiding (rec; elim)
+
+open import Cubical.Categories.Category.Base
+open import Cubical.Categories.Displayed.Base
+open import Cubical.Categories.Displayed.Functor
+open import Cubical.Categories.Functor.Base
+
+open import Cubical.Categories.Displayed.Section.Base
+open import Cubical.Categories.Displayed.Constructions.Weaken as Weaken
+open import Cubical.Categories.Displayed.Constructions.Reindex as Reindex
+open import Cubical.Categories.Displayed.Instances.Path as PathC
+
+private
+  variable
+    ℓC ℓC' ℓD ℓD' ℓE ℓE' ℓCᴰ ℓCᴰ' ℓDᴰ ℓDᴰ' : Level
+
+open Category
+open Categoryᴰ
+open Functor
+open Functorᴰ
+open Section
+
+module _ (C : Category ℓC ℓC')(D : Category ℓD ℓD') where
+  private
+    ⊎Ob = C .ob Sum.⊎ D .ob
+
+    Hom⊎ : ⊎Ob → ⊎Ob → Type (ℓ-max ℓC' ℓD')
+    Hom⊎ (inl c) (inl c') = Lift {j = ℓD'} $ C [ c , c' ]
+    Hom⊎ (inr d) (inr d') = Lift {j = ℓC'} $ D [ d , d' ]
+    Hom⊎ (inl c) (inr d') = ⊥*
+    Hom⊎ (inr d) (inl c') = ⊥*
+
+    -- the following inductive definition is isomorphic and very
+    -- slightly more ergonomic, but lives at a higher universe level
+    data Hom⊎' : ⊎Ob → ⊎Ob → Type (ℓ-max (ℓ-max ℓC ℓD) (ℓ-max ℓC' ℓD')) where
+      inl : ∀ {c c'} → C [ c , c' ] → Hom⊎' (inl c) (inl c')
+      inr : ∀ {d d'} → D [ d , d' ] → Hom⊎' (inr d) (inr d')
+
+  _⊎_ : Category (ℓ-max ℓC ℓD) (ℓ-max ℓC' ℓD')
+  _⊎_ .ob = ⊎Ob
+  _⊎_ .Hom[_,_] = Hom⊎
+  _⊎_ .id {inl c} = lift $ C .id
+  _⊎_ .id {inr d} = lift $ D .id
+  _⊎_ ._⋆_ {inl c} {inl c'} {inl c''} f g = lift (f .lower ⋆⟨ C ⟩ g .lower )
+  _⊎_ ._⋆_ {inr d} {inr d'} {inr d''} f g = lift (f .lower ⋆⟨ D ⟩ g .lower)
+  _⊎_ .⋆IdL {inl _} {inl _} f = cong lift (C .⋆IdL (f .lower))
+  _⊎_ .⋆IdL {inr _} {inr _} f = cong lift (D .⋆IdL (f .lower))
+  _⊎_ .⋆IdR {inl _} {inl _} f = cong lift (C .⋆IdR (f .lower))
+  _⊎_ .⋆IdR {inr _} {inr _} f = cong lift (D .⋆IdR (f .lower))
+  _⊎_ .⋆Assoc {inl _} {inl _} {inl _} {inl _} f g h =
+    cong lift (C .⋆Assoc (f .lower) (g .lower) (h .lower))
+  _⊎_ .⋆Assoc {inr _} {inr _} {inr _} {inr _} f g h =
+    cong lift (D .⋆Assoc (f .lower) (g .lower) (h .lower))
+  _⊎_ .isSetHom {inl _} {inl _} = isOfHLevelLift 2 (C .isSetHom)
+  _⊎_ .isSetHom {inr _} {inr _} = isOfHLevelLift 2 (D .isSetHom)
+  _⊎_ .isSetHom {inl _} {inr _} = isProp→isSet isProp⊥*
+  _⊎_ .isSetHom {inr _} {inl _} = isProp→isSet isProp⊥*
+
+module _ {C : Category ℓC ℓC'}{D : Category ℓD ℓD'} where
+  -- TODO: should these be inl and inr?
```

Comment:
Why would it not be?

Reply:
I meant should they be capitalized or not